xbout/load.py

```diff
@@ -199,7 +201,7 @@ def _expand_filepaths(datapath):
              " `file_cache_maxsize` global option to {} to accommodate this. "
              "Recommend using `xr.set_options(file_cache_maxsize=NUM)`"
              " to explicitly set this to a large enough value."
-             .format(str(len(filepaths))), UserWarning)
+             .format(str(len(filepaths))), category=xBOUTWarning)
```

Comment:
I'm not sure that an `xBOUTWarning` is more helpful than a `UserWarning` here. Defining your own errors or warnings is supposed to make new types of failure clearer. A generic `xBOUTWarning` doesn't tell us anything new because the warning traceback will already show the error was raised from within xBOUT. 

I would say we should either use python generic warnings or make even more specific types of warnings.

Reply:
`UserWarning` is the default raised by `warnings.warn()`, so my thinking was that maybe we don't want to ignore all `UserWarning`s, because they might (although possibly shouldn't) be raised by external libraries, whereas there are warnings raised by `xBOUT` for information to the user that we can happily ignore in the test suit. Maybe something more specific would be better though - `xBOUTInfo` maybe for things like this https://github.com/boutproject/xBOUT/blob/212f9137c9f4e37a588b35bd75b1eefb4c6a6fc5/xbout/load.py#L132-L134
but something more serious - `RuntimeWarning`? `xBOUTWarning`? - for things like this https://github.com/boutproject/xBOUT/blob/212f9137c9f4e37a588b35bd75b1eefb4c6a6fc5/xbout/load.py#L199-L205

Reply:
We could literally make them specific: `GeometryWarning` and `FileCacheWarning`. Then we don't have the unecessary `xBOUT prefix`, catching them won't ignore `UserWarnings`, they still tell you about what's gone wrong, and they are general enough that there might be other sensible places to also raise a `GeometryWarning` with a different message.

Reply:
Coming back to look at this again, I think sticking with UserWarning may be the best thing. My original aim was to filter the warnings, but the filter can be set on the message as well as the warning type. I'm going to get rid of the custom warning types now...